tool/e2e-tests/prescribe/prescriptionPagination.ts

```diff
@@ -1,63 +1,37 @@
-async function getNextPrescriptionId(driver: ThenableWebDriver) {
-  return await (await driver.manage().getCookie("Next-Prescription-Id")).value
-}
+    await loadSupplierTestPack1Examples(driver)
 
-async function checkPageIsShowingCurrentPrescription(driver: ThenableWebDriver, startingPrescriptionId: string) {
-  expect(await driver.getCurrentUrl()).toContain(`prescribe/edit?prescription_id=${encodeURIComponent(startingPrescriptionId)}`)
-  await driver.wait(until.elementLocated(sendPageTitle), defaultWaitTimeout)
-}
+    let paginationItemElements = await driver.findElements(By.className("pagination-item"))
+    const startingPageElement = paginationItemElements[1]
+    await checkPageIsShowingCurrentPrescription(driver, startingPageElement)
 
-async function loadTheNextPrescription(driver: ThenableWebDriver) {
-  await driver.actions({async: true}).sendKeys(Key.ARROW_RIGHT).perform()
-}
+    let nextPageElement = paginationItemElements[2]
+    await loadTheNextPrescription(driver, nextPageElement)
 
-async function checkPageIsShowingNextPrescription(driver: ThenableWebDriver, nextPrescriptionId: string) {
-  await driver.wait(until.urlContains(`prescribe/edit?prescription_id=${encodeURIComponent(nextPrescriptionId)}`))
-  await driver.wait(until.elementLocated(sendPageTitle))
-}
+    // pagniation has refreshed the dom, re-find elements for pagination so they're not stale
+    paginationItemElements = await driver.findElements(By.className("pagination-item"))
+    nextPageElement = paginationItemElements[2]
 
-async function getPreviousPrescriptionId(driver: ThenableWebDriver) {
-  return await (await driver.manage().getCookie("Previous-Prescription-Id")).value
-}
+    await checkPageIsShowingNextPrescription(driver, nextPageElement)
+  })
+})
 
-async function loadThePreviousPrescription(driver: ThenableWebDriver) {
-  await driver.actions({async: true}).sendKeys(Key.ARROW_LEFT).perform()
+async function checkPageIsShowingCurrentPrescription(driver: ThenableWebDriver, startingPageElement: WebElement) {
+  expect((await startingPageElement.getAttribute("class")).split(" ").indexOf("selected") >= 0).toBeTruthy()
 }
 
-async function checkPageIsShowingThePreviousPrescription(driver: ThenableWebDriver, previousPrescriptionId: string) {
-  await driver.wait(until.urlContains(`prescribe/edit?prescription_id=${encodeURIComponent(previousPrescriptionId)}`))
-  await driver.wait(until.elementLocated(sendPageTitle))
+async function loadTheNextPrescription(driver: ThenableWebDriver, nextPageElement: WebElement) {
+  await nextPageElement.click()
 }
 
-function checkPreviousPrescriptionIsTheStartingPrescription(previousPrescriptionId: string, startingPrescriptionId: string) {
-  expect(previousPrescriptionId).toEqual(startingPrescriptionId)
+async function checkPageIsShowingNextPrescription(driver: ThenableWebDriver, nextPageElement: WebElement) {
+  expect((await nextPageElement.getAttribute("class")).split(" ").indexOf("selected") >= 0).toBeTruthy()
```

Comment:
If I'm honest this expect and the one on line 28 have me a little confused about what they are actually checking

Reply:
It's checking against a pagination list to ensure that elements are selected as they should be, so this test ensures that we can navigate from page 1 to page 2 successfully

Reply:
Great thank you